File: Array/fanbei/0630/java/Solution06_30.java
package com.example.lsn37_dagger2_subcomponent;

public class Solution06_30 {

    class ListNode{
        int val;
        ListNode next;
        ListNode(int x){
            this.val = x;
            this.next = null;
        }
    }

    public ListNode reverseList(ListNode head){
        if(head == null){
            return head;
        }
        ListNode newHead = null;
        ListNode next = head;
        ListNode temp = null;
        while(next != null){
            temp = next.next;
            next.next = newHead;
            newHead = next;
            next = temp;
        }
        return newHead;
    }

}

File: Array/fanbei/0630/c/solution06_30.c
#include <stdlib.h>
#include <stdio.h>

struct ListNode{
    int val;
    struct ListNode* next;
};

struct ListNode* reverseList(struct ListNode* node){
    if (node == NULL){
        return NULL;
    }
    struct  ListNode* newHead = NULL;
    struct  ListNode* head = node;
    struct ListNode* tmp = NULL;
    while (head!=NULL){
        tmp = head->next;
        head->next = newHead;
        newHead = node;
        node = tmp;
    }
    return newHead;
}

